tornado/ioloop.py

```diff
@@ -55,9 +55,36 @@
 from tornado import stack_context
 from tornado.util import (
     PY3, Configurable, errno_from_exception, timedelta_to_seconds,
-    TimeoutError, unicode_type, import_object, xrange,
+    TimeoutError, unicode_type, import_object, xrange as _xrange,
 )
 
+from tornado.stack_context import wrap
+
+try:
+    import cython
+except ImportError:
+    # Shadow cython implementation for decoration
+    class _cython:
+        compiled = False
+        double = float
+        def _ident(*p, **kw):
+            def d(f):
+                return f
+            return d
+        locals().update(dict(
+            locals = _ident
+        ))
+    globals().update(dict(
+        cython = _cython
+    ))
+
+if PY3:
```

Comment:
You are already checking for PY3 on line 71. Couldn't you that there so we have only one place for this kind of things?